ui/page/main_page.go

```diff
@@ -497,7 +540,7 @@ func (mp *MainPage) popToFragment(pageID string) {
 // to be eventually drawn on screen.
 // Part of the load.Page interface.
 func (mp *MainPage) Layout(gtx layout.Context) layout.Dimensions {
-	mp.UpdateBalance(false)
+	mp.updateExchangeSetting()
```

Comment:
Let's add a comment here to give future code maintainers an idea why this is being done each time `Layout` is called.
```suggestion
	mp.updateExchangeSetting() // the setting may have changed, leading to this window refresh, let's check
```